### PR TITLE
mimic: 1.2.0.2 -> 1.3.0.1

### LINKED ISSUE #137373: Mimic (Mycroft's TTS engine)
### Describe the bug

I want to use the TTS Mimic on a Debian 11. After installation, I Tried reading a text.
Cli usage give me an "alsa lib error".

### Steps To Reproduce
1. nix-env -iA nixpkgs.mimic
2. mimic -t "hello world"
```console
ALSA lib conf.c:3725:(snd_config_hooks_call) Cannot open shared library libasound_module_conf_pulse.so (/nix/store/anvz12gs16zzfwcj42fnzkg70wj5v3kx-alsa-lib-1.2.4/lib/alsa-lib/libasound_module_conf_pulse.so: cannot open shared object file: No such file or directory)
ALSA lib pcm.c:2731:(snd_pcm_open_noupdate) Unknown PCM default
audio_open_alsa: failed to open audio device default. No such file or directory
```

### Notify maintainers
@noneucat

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
these paths will be fetched (48.64 MiB download, 222.38 MiB unpacked):
  /nix/store/5qjycalzb9sqzvqg65kf5zimqwjabm9g-gcc-10.3.0
  /nix/store/5wvf6qy2pjijd93n4dn6vvbn45ij9fdr-linux-headers-5.12
  /nix/store/60kcip377kqab1p7ks3q9skvzm785ljg-bash-interactive-4.4-p23-doc
  /nix/store/74kv08wjf06ifgk4dxrnra4qhzr5s1w4-gcc-wrapper-10.3.0
  /nix/store/7gdji02y6rbnzv5mcxcf3mwksk785y8j-stdenv-linux
  /nix/store/7wy23fyz7lszd7izsla09d5bzmc57x22-gnumake-4.3
  /nix/store/7ypcjsla15p7l1bx3dlnlsx708n293rd-expand-response-params
  /nix/store/86zys752g7fg2mwad7yhll285y9w3lym-patchelf-0.12
  /nix/store/9r0a3dipi8saq2zasp668zsk6qhqp5jb-glibc-2.32-48-dev
  /nix/store/ccb8104ais1q6mbzq98vsn5lmj2wp88f-findutils-4.7.0
  /nix/store/d6892i796730p9iggkh1xyaayiymwn98-gawk-5.1.0
  /nix/store/dn6pahjvv6b9rxdplw5jfpw8lnnyawl4-ed-1.17
  /nix/store/fs4mvc0f27vhf4jv6hy54vmj7m7m8lqb-nix-info
  /nix/store/hzr3gfl4p6lriqiwqbd27yvmjwkzvbx2-diffutils-3.7
  /nix/store/lkk01rbdq39l29k9yx8fa0g8p0fhvvyn-binutils-wrapper-2.35.1
  /nix/store/nhf3114vx6v0gih1gpm3cg3fkyhnclac-gnused-4.8
  /nix/store/nzgcf0y8a7gc4rjq01nc7x8656sz5nsf-bash-interactive-4.4-p23-man
  /nix/store/r1jlizxxdpydhhfmza0sidjjrggf0i4w-bash-interactive-4.4-p23-info
  /nix/store/sgrwh6jsrshrnicqi67fck336q6ldh2d-binutils-2.35.1
  /nix/store/v7wbjl8gll2xkgnd3im55mddsd118ih8-patch-2.7.6
  /nix/store/wzach25vwapl3sa6f7gylqank3jhhsin-bash-interactive-4.4-p23-dev
copying path '/nix/store/60kcip377kqab1p7ks3q9skvzm785ljg-bash-interactive-4.4-p23-doc' from 'https://cache.nixos.org'...
copying path '/nix/store/wzach25vwapl3sa6f7gylqank3jhhsin-bash-interactive-4.4-p23-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/r1jlizxxdpydhhfmza0sidjjrggf0i4w-bash-interactive-4.4-p23-info' from 'https://cache.nixos.org'...
copying path '/nix/store/nzgcf0y8a7gc4rjq01nc7x8656sz5nsf-bash-interactive-4.4-p23-man' from 'https://cache.nixos.org'...
copying path '/nix/store/sgrwh6jsrshrnicqi67fck336q6ldh2d-binutils-2.35.1' from 'https://cache.nixos.org'...
copying path '/nix/store/hzr3gfl4p6lriqiwqbd27yvmjwkzvbx2-diffutils-3.7' from 'https://cache.nixos.org'...
copying path '/nix/store/dn6pahjvv6b9rxdplw5jfpw8lnnyawl4-ed-1.17' from 'https://cache.nixos.org'...
copying path '/nix/store/7ypcjsla15p7l1bx3dlnlsx708n293rd-expand-response-params' from 'https://cache.nixos.org'...
copying path '/nix/store/ccb8104ais1q6mbzq98vsn5lmj2wp88f-findutils-4.7.0' from 'https://cache.nixos.org'...
copying path '/nix/store/d6892i796730p9iggkh1xyaayiymwn98-gawk-5.1.0' from 'https://cache.nixos.org'...
copying path '/nix/store/7wy23fyz7lszd7izsla09d5bzmc57x22-gnumake-4.3' from 'https://cache.nixos.org'...
copying path '/nix/store/nhf3114vx6v0gih1gpm3cg3fkyhnclac-gnused-4.8' from 'https://cache.nixos.org'...
copying path '/nix/store/5wvf6qy2pjijd93n4dn6vvbn45ij9fdr-linux-headers-5.12' from 'https://cache.nixos.org'...
copying path '/nix/store/fs4mvc0f27vhf4jv6hy54vmj7m7m8lqb-nix-info' from 'https://cache.nixos.org'...
copying path '/nix/store/9r0a3dipi8saq2zasp668zsk6qhqp5jb-glibc-2.32-48-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/v7wbjl8gll2xkgnd3im55mddsd118ih8-patch-2.7.6' from 'https://cache.nixos.org'...
copying path '/nix/store/lkk01rbdq39l29k9yx8fa0g8p0fhvvyn-binutils-wrapper-2.35.1' from 'https://cache.nixos.org'...
copying path '/nix/store/5qjycalzb9sqzvqg65kf5zimqwjabm9g-gcc-10.3.0' from 'https://cache.nixos.org'...
copying path '/nix/store/86zys752g7fg2mwad7yhll285y9w3lym-patchelf-0.12' from 'https://cache.nixos.org'...
copying path '/nix/store/74kv08wjf06ifgk4dxrnra4qhzr5s1w4-gcc-wrapper-10.3.0' from 'https://cache.nixos.org'...
copying path '/nix/store/7gdji02y6rbnzv5mcxcf3mwksk785y8j-stdenv-linux' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.0-8-amd64, Debian GNU/Linux, 11 (bullseye)`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.15`
 - channels(jferry): `"nixpkgs-21.05.2734.74d017edb67"`
 - nixpkgs: `/home/jferry/.nix-defexpr/channels/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

### PR DESCRIPTION
**Pull request by wamserma**
###### Motivation for this change

* version bump
* adopt upstream renaming of repository
* add wrapper for alsaPlugins
* Closes https://github.com/NixOS/nixpkgs/issues/137373

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review pr 137816"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/mimic/default.nix [hunk 1]
```diff
@@ -1,29 +1,37 @@
-{ config, lib, stdenv, autoreconfHook, fetchFromGitHub, pkg-config
-, alsa-lib, libtool, icu
+{ config, lib, stdenv, autoreconfHook, fetchFromGitHub, pkg-config, makeWrapper
+, alsa-lib, alsaPlugins, libtool, icu, pcre2
```
Comment (bobby285271) [inline pkgs/applications/audio/mimic/default.nix:2]:
```suggestion
, alsa-lib, alsa-plugins, libtool, icu, pcre2
```

File: pkgs/applications/audio/mimic/default.nix [hunk 2]
```diff
@@ -1,29 +1,37 @@
...
 
   nativeBuildInputs = [
     autoreconfHook
     pkg-config
+    makeWrapper
   ];
 
   buildInputs = [
     alsa-lib
+    alsaPlugins
```
Comment (bobby285271) [inline pkgs/applications/audio/mimic/default.nix:24]:
```suggestion
    alsa-plugins
```

File: pkgs/applications/audio/mimic/default.nix [hunk 3]
```diff
@@ -1,29 +1,37 @@
...
     libtool
     icu
+    pcre2
   ] ++ lib.optional pulseaudioSupport libpulseaudio;
 
+  postInstall = ''
+    wrapProgram $out/bin/mimic \
+      --run "export ALSA_PLUGIN_DIR=${alsaPlugins}/lib/alsa-lib"
```
Comment (bobby285271) [inline pkgs/applications/audio/mimic/default.nix:32]:
```suggestion
      --run "export ALSA_PLUGIN_DIR=${alsa-plugins}/lib/alsa-lib"
```

Comment (wamserma) [reply pkgs/applications/audio/mimic/default.nix]: already updated - funnily local build worked nevertheless

**Review by mothsART — APPROVED** [review]

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 137816` at 827b7b96 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>mimic</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

libtool is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/applications/audio/mimic/default.nix:22:3:
```
   |
22 |   buildInputs = [
   |   ^
```
</li>
  </ul>
</details>

**Comment by noneucat** [comment]
@wamserma It might not be necessary to write our own wrapper, I think   `alsaPluginWrapper` already does this (https://github.com/NixOS/nixpkgs/blob/master/pkgs/os-specific/linux/alsa-project/alsa-plugins/wrapper.nix)

**Review by fx-chun — APPROVED** [review]

**Comment by wamserma** [comment]
> @wamserma It might not be necessary to write our own wrapper, I think `alsa-plugins-wrapper` already does this. 
Writing the custom wrapper was quicker than checking if https://github.com/NixOS/nixpkgs/issues/6860#issuecomment-615879126 was resolved. :D

**Comment by wamserma** [comment]
@SuperSandro2000 can you do the merge?

File: pkgs/applications/audio/mimic/default.nix [hunk 4]
```diff
@@ -1,29 +1,37 @@
...
 
 stdenv.mkDerivation rec {
   pname = "mimic";
-  version = "1.2.0.2";
+  version = "1.3.0.1";
 
   src = fetchFromGitHub {
     rev = version;
-    repo = "mimic";
+    repo = "mimic1";
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/mimic/default.nix:11]: Both are 404.

Comment (wamserma) [reply pkgs/applications/audio/mimic/default.nix:11]: https://github.com/MycroftAI/mimic1/ works for me.

**Pull request merged**